sdcm/utils/decorators.py

```diff
@@ -155,6 +158,96 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+class LatencyCalculatorMixedResult(GenericResultTable):
+    class Meta:
+        name = ""  # to be set by the decorator
+        description = ""
+        Columns = [
+            ColumnMetadata(name="P90 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P90 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="duration", unit="HH:MM:SS", type=ResultType.DURATION)
+        ]
+
+
+class LatencyCalculatorWriteResult(GenericResultTable):
+    class Meta:
+        name = ""  # to be set by the decorator
+        description = ""
+        Columns = [
+            ColumnMetadata(name="P90 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="duration", unit="HH:MM:SS", type=ResultType.DURATION)
+        ]
+
+
+class LatencyCalculatorReadResult(GenericResultTable):
+    class Meta:
+        name = ""  # to be set by the decorator
+        description = ""
+        Columns = [
+            ColumnMetadata(name="P90 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="duration", unit="HH:MM:SS", type=ResultType.DURATION)
+        ]
+
+
+class ReactorStallStatsResult(GenericResultTable):
+    class Meta:
+        name = ""
+        description = ""
+        Columns = [
+            ColumnMetadata(name="total", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="10ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="20ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="30ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="50ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="100ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="200ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="1000ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="2000ms", unit="count", type=ResultType.INTEGER),
+        ]
+
+
+workload_to_table = {
+    "mixed": LatencyCalculatorMixedResult,
+    "write": LatencyCalculatorWriteResult,
+    "read": LatencyCalculatorReadResult
+}
+
+
+def send_result_to_argus(argus_client: ArgusClient, workload: str, name: str, description: str, cycle: int, result: dict):
```

Comment:
this is not a decorator, so me might put it in a diffrent module ?

Reply:
it's closely related to this one, I'm not sure. 
maybe along with tables above would make sense.